### PR TITLE
[QA-864] feat: Create openapi to personal tokens

### PR DESCRIPTION
**Pull request by JoiceFelix**

### CODE REVIEW

**Comment by JoiceFelix** [comment]
I have read the CLA Document and I hereby sing the CLA

**Comment by PatrickMenoti** [comment]
recheck

**Comment by JoiceFelix** [comment]
> recheck


I have read the CLA Document and I hereby sing the CLA

**Comment by JoiceFelix** [comment]
recheck 

I have read the CLA Document and I hereby sing the CLA

**Comment by JoiceFelix** [comment]
recheck

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
+#Python
+python/
+# Environments
+.env
+.venv
+env/
+venv/
+ENV/
+env.bak/
+venv.bak/
```
Comment (vinigfer) [inline .gitignore:15]:
```suggestion
*env*
*ENV*
```

**Comment by JoiceFelix** [comment]
I have read the CLA Document and I hereby sing the CLA

File: personal_tokens.yaml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+tags:
+  - name: Personal Token
+    description: List all personal tokens
+paths:
+  /iam/personal_tokens:
+    get:
+      tags:
+        - Personal Token
+      summary: List an existing personal token 
+      description: List an existing personal token
```
Comment (vinigfer) [inline personal_tokens.yaml:27]:
```suggestion
      summary: List all existing personal token 
      description: List all existing personal token
```

File: .gitignore (hunk 1 above)
Comment (JoiceFelix) [reply .gitignore]: Done! https://github.com/aziontech/azionapi-openapi/pull/65/commits/357ddb698be20531507d1b1bfe07240a06705289

File: personal_tokens.yaml (hunk 2 above)
Comment (JoiceFelix) [reply personal_tokens.yaml]: done! https://github.com/aziontech/azionapi-openapi/pull/65/commits/ea37f50019e92f8fd2deb425f2e51153b7decec7

File: personal_tokens.yaml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+              $ref: '#/components/schemas/ResponseGet'
+              
+            example:
+              - uuid: "s6006r20-5lf8-4750-a5c4-2820552643c0"
+                name: "Personal Token 1"
+                expires_at: "2023-04-12T23:59:59Z"
+                description: "my personal token"
+              - uuid: "4if9f6rd-5q7f-74dd-s856-451e5ef4048b"
+                name: "Personal Token 2" 
+                expires_at: "2023-04-12T23:59:59Z"
```
[the remark this answers is not in the document]
Comment (JoiceFelix) [reply personal_tokens.yaml:77]: No, but I was relying on the azion documents.

[the remark this answers is not in the document]
Comment (vinigfer) [reply personal_tokens.yaml:77]: Never trust anything/anyone ... not even me 😅 
If `created` is a property that returns in the response, please add it to the examples.

**Comment by JoiceFelix** [comment]
recheck

**Comment by JoiceFelix** [comment]
I have read the CLA Document and I hereby sing the CLA

File: personal_tokens.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ftoldato) [reply personal_tokens.yaml:77]: My suggestion is to look at the API return, via curl or postman. Add `created` in the example.

```
Return postman.
{
            "uuid": "",
            "name": "",
            "created": "2023-06-12T11:50:48.263377Z",
            "expires_at": "2024-06-11T23:59:59Z",
            "description": ""
}
```

File: .gitignore [hunk 4]
```diff
@@ -1,4 +1,10 @@
...
 package-lock.json
-package.json
\ No newline at end of file
+package.json
+
+#Python
+python/
+# Environments
+*env*
+*ENV*
```
Comment (alisson-marcelo-azion) [inline .gitignore:10]: New line at the end of the file.

I believe ENV is not needed here and venv is also recommended 😉 .

File: personal_tokens.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (alisson-marcelo-azion) [reply personal_tokens.yaml:77]: My suggestion is to remove all examples and add them in another PR. This is the first PR and examples are only needed when and if openapi is the basis of our documentation.

Please do not consume time to define examples now.

File: personal_tokens.yaml [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+              nullable: true
+              example: null
+            previous:
+              type: string
+              nullable: true
+              example: null
+        results:
+            type: array
+            items:
+              $ref: '#/components/schemas/ResponseGet'
```
Comment (alisson-marcelo-azion) [inline personal_tokens.yaml:68]: Please replace by:
```
              $ref: '#/components/schemas/PersonalTokenResponseGet'
```

File: personal_tokens.yaml [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                $ref: '#/components/schemas/PersonalTokenResponseWithResults'          
+components:
+  schemas:
+    PersonalTokenResponseWithResults:
+      type: object
+      properties:
+        count:
+          type: integer
+          format: int64
+          example: 2
```
Comment (alisson-marcelo-azion) [inline personal_tokens.yaml:45]: Remove all examples. In this case, you are providing an integer as an example of an integer. It's so obvious that we shouldn't consume file space with it.

File: personal_tokens.yaml (hunk 5 above)
Comment (alisson-marcelo-azion) [reply personal_tokens.yaml:68]: This way, the merge of multiple files is possible as they will not have conflicting names.

**Comment by alisson-marcelo-azion** [comment]
Please add a comment or change the PR title to explicit that what you are delivering here is the first version of the specification with just the _get personal tokens list_ operation.

File: .gitignore (hunk 4 above)
Comment (vinigfer) [reply .gitignore:10]: `*env*` already contains `venv` right? 🤔

File: personal_tokens.yaml (hunk 5 above)
Comment (JoiceFelix) [reply personal_tokens.yaml:68]: tks, for suggestion =D

Comment (JoiceFelix) [reply personal_tokens.yaml]: done! https://github.com/aziontech/azionapi-openapi/pull/65/commits/48c433b8e1bdbc9e2790f8a723042f737e9bc42f

File: personal_tokens.yaml (hunk 6 above)
Comment (JoiceFelix) [reply personal_tokens.yaml]: done!https://github.com/aziontech/azionapi-openapi/pull/65/commits/10f48c580e73f9dc2a6a71cdb25d371166cfa7ee

File: personal_tokens.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (JoiceFelix) [reply personal_tokens.yaml]: done!https://github.com/aziontech/azionapi-openapi/pull/65/commits/10f48c580e73f9dc2a6a71cdb25d371166cfa7ee

**Review by vinigfer — APPROVED** [review]

File: personal_tokens.yaml [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+          type: string
+  securitySchemes:
+    tokenAuth:
+      type: apiKey
+      name: Authorization
+      in: header
+      description: |
+        You must inform a token to auth.
+        Usage format: `Token <TOKEN>`
+
```
Comment (ftoldato) [inline personal_tokens.yaml:90]: remove line end

File: .gitignore (hunk 4 above)
Comment (ftoldato) [inline .gitignore:10]: remove line end

Comment (alisson-marcelo-azion) [reply .gitignore:10]: yeah, my bad.

Comment (alisson-marcelo-azion) [reply .gitignore:10]: It's opposite @ftoldato , GitHub is asking to add the line.

File: personal_tokens.yaml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+        description: 
+          type: string
+  securitySchemes:
+    tokenAuth:
+      type: apiKey
+      name: Authorization
+      in: header
+      description: |
+        You must inform a token to auth.
+        Usage format: `Token <TOKEN>`
```
Comment (alisson-marcelo-azion) [inline personal_tokens.yaml:89]: Line at the end

**Review by alisson-marcelo-azion — CHANGES_REQUESTED** [review]

File: personal_tokens.yaml (hunk 8 above)
Comment (ftoldato) [reply personal_tokens.yaml:89]: bad sugestion, I understand to remove end line.

Comment (JoiceFelix) [reply personal_tokens.yaml:89]: This was done on! https://github.com/aziontech/azionapi-openapi/pull/65/commits/2d7b8445b0fba6686bfa9da00cf8c5704a68854a

File: personal_tokens.yaml (hunk 7 above)
Comment (JoiceFelix) [reply personal_tokens.yaml]: Done! https://github.com/aziontech/azionapi-openapi/pull/65/commits/2d7b8445b0fba6686bfa9da00cf8c5704a68854a

**Review by ftoldato — DISMISSED** [review]

File: .gitignore (hunk 4 above)
Comment (alisson-marcelo-azion) [reply .gitignore:10]: @JoiceFelix if GitHub place the simbol ⊖ at the end of the file, it means that the file end of the file is wrong. So, keep one and just one line at the end of the file.

**Review by alisson-marcelo-azion — CHANGES_REQUESTED** [review]
Problemas with EOF

**Comment by JoiceFelix** [comment]
> Problemas with EOF

This was done on! https://github.com/aziontech/azionapi-openapi/commit/2d7b8445b0fba6686bfa9da00cf8c5704a68854a

File: personal_tokens.yaml [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+              type: string
+              nullable: true
+            previous:
+              type: string
+              nullable: true
+        results:
+            type: array
+            items:
+              $ref: '#/components/schemas/PersonalTokenResponseGet'
+
```
Comment (alisson-marcelo-azion) [inline personal_tokens.yaml:64]: We don't describe any xml as return value in any other openapi specification, so remove this key _xml_ from here

Comment (JoiceFelix) [reply personal_tokens.yaml:64]: Done!https://github.com/aziontech/azionapi-openapi/pull/65/commits/1820bed0618b37d6dc612e687613cb918d4880d2

**Comment by JoiceFelix** [comment]
> Add blank line at the end of file

This was done on ! https://github.com/aziontech/azionapi-openapi/pull/65/commits/2174b659624773845252394368c9c80f48635b7b

**Review by ftoldato — APPROVED** [review]

**Review by alisson-marcelo-azion — APPROVED** [review]

**Pull request merged**